Feat/coffeechat 시간 변경 및 예정된 커피챗 목록 조회 기능 추가

geonuu09 opened the pull request:
### 변경사항

AS-IS
- 예정된 커피챗 신청에 대해서 조회하는 기능 없음

- 커피챗 등록한 시간 변경 불가능

TO-BE
- 커피챗 신청 후 멘토가 수락 시 커피챗 예정됨 → 예정된 커피챗 목록 조회 가능

- 커피챗 시간 변경 가능

- 기존 시간 + 업데이트 요청 시간 비교 후 추가 및 삭제

기타 변경사항
- mento → mentor 오타 수정 및 통일

- 클래스마다 책임 분리

- 페이지네이션 구조 (data + pagination)로 응답을 변환하여 API 응답을 일관성 있게 개선

- Pageable을 활용한 기존 응답을 팀 컨벤션에 맞게 pagination 구조로 변경

- collect 사용 시 toList() 적용하여 가독성 개선

- NotNull 어노테이션 추가로 유효성 검사 강화

- CoffeeChatTimeRequestDto → CoffeeChatTimeListDto로 클래스 네이밍 변경

불필요한 공통 메서드 정리 및 코드 리팩토링 진행
### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [x] API 테스트 


- 예정된 커피챗 조회 
![image](https://github.com/user-attachments/assets/66b24529-978c-4c8d-b2e6-6c9d40923cda)


- 시간 변경 API (PK 변동 x) 
![image](https://github.com/user-attachments/assets/41c031db-44bc-4937-9594-7ef5b7a007b5)

geonuu09 committed:
feat & refactor : 확정된 커피챗 조회 API 기능 추가 및 책임에 따른 클래스 분리

geonuu09 committed:
fix: Long 타입 비교 오류 수정 (`!=` -> `equals()`)

geonuu09 committed:
refactor: 람다 표현식을 메서드 참조로 변경하여 가독성 개선

geonuu09 committed:
Merge branch 'develop' of https://github.com/BootTalk-Service/boot-talk-back into feat/coffeechat

# Conflicts:
#	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/controller/CoffeeChatInfoController.java
#	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/repository/CoffeeChatApplicationRepository.java
#	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatApplicationService.java
#	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatInfoService.java

geonuu09 committed:
refactor: 책임에 맞게 클래스 분류

geonuu09 committed:
refactor: userId -> mentoId 필드명 변경으로 인한 수정

geonuu09 committed:
refactor: userId -> mentoId 필드명 변경으로 인한 수정

geonuu09 committed:
feat : 수락된 커피챗 목록 조회 API 구현 완료

geonuu09 committed:
rename : CoffeeChatTimeRequestDto -> CoffeeChatTimeListDto 클래스명 변경

geonuu09 committed:
refactor : NotNull 어노테이션을 추가해서 유효성 검사

geonuu09 committed:
refactor : 멘토 아이디 필드 추가

geonuu09 committed:
feat : 커피챗 시간 수정 API 추가

geonuu09 committed:
refactor : 공통 로직 메소드

geonuu09 committed:
refactor : 공통 로직 메소드

geonuu09 committed:
refactor : 코드 수정  collect -> toList() 사용

geonuu09 committed:
Merge branch 'develop' of https://github.com/BootTalk-Service/boot-talk-back into feat/coffeechat

geonuu09 committed:
refactor : PageNation 적용

geonuu09 committed:
refactor : PageNation 적용

geonuu09 committed:
rename : mento -> mentor 오타수정

geonuu09 committed:
Merge branch 'develop' into feat/coffeechat

jooyeon0907 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatInfoService.java:
```diff
@@ -79,8 +64,8 @@ public void deleteMyCoffeeChatInfo(Long userId) {
-    protected  CoffeeChatInfo getCoffeeChatInfoByUserId(Long userId) {
-        return coffeeChatInfoRepository.findByMento_UserId(userId)
-            .orElseThrow(() -> new CustomException(ErrorCode.USER_COFFEE_CHAT_NOT_FOUND));
+    private CoffeeChatInfo getCoffeeChatInfoByUserId(Long userId) {
```
CoffeeChatReceivedService 에서 해당 메소드 사용하여 protected 로 변경했었습니다..!

geonuu09 replied:
아하 그런가요! 저는 변경하지 않았지만 충돌이 일어나서 private를 선택했나봐요! 수정하겠습니다~

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/controller/CoffeeChatApplicationController.java:
```diff
@@ -45,14 +45,23 @@ public ResponseEntity<List<CoffeeChatApplicationResponseDto>> getMyCoffeeChatApp
         return ResponseEntity.ok(coffeeChatAppService.getMyCoffeeChatApps(userId));
```
여기 위에 getMyCoffeeChatApps는 추후에 페이징 처리하나요 List로 사용하나요??

geonuu09 replied:
이 부분은 주연님 파트여서 데일리 스크럼때 논의해보면 좋을 것 같습니다!

shCHO9801 reviewed (commented):
많은 내용들 다루느라 고생하셨습니다 :D
제가 작성한 코멘트들 한번씩 확인해주세요!

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatTimeService.java:
```diff
@@ -46,23 +45,60 @@ public List<CoffeeChatTimeResponseDto> createCoffeeChatTimes(Long userId,
+        // 기존과 비교 후 새롭게 추가된 시간을 toAdd 리스트에 추가
+        List<CoffeeChatTime> toAdd = newTimeSlots.stream()
+            .filter(newSlot -> existingTimeSlots.stream()
+                .noneMatch(existing -> existing.equals(newSlot)))
```
제가 stream을 정확히 잘 이해는 못하고 있기는 한데
CoffeeChatTime 객체간의 비교를 하시는 걸로 보입니다
.equals 메소드가 각 모든 필드를 다 비교하는지 궁금합니다

jooyeon0907 reviewed (commented):
주석 잘 작성해 주셔서 코드 리뷰하는 데 도움 되었습니다
오타 수정해 주시느라 고생하셨습니다..!

jooyeon0907 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatInfoApprovedDto.java:
```diff
@@ -0,0 +1,33 @@
+import com.icandoit.boottalk.coffeeChat.entity.enums.StatusType;
+import java.time.LocalDateTime;
+
+public record CoffeeChatInfoApprovedDto(
```
CoffeeChatApplicationResponseDto 와 반환값들이 일치하는 것 같은데 따로 CoffeeChatInfoApprovedDto 를 분리한 이유가 궁금합니다!

geonuu09 replied:
`CoffeeChatApplicationResponseDto`와 `CoffeeChatInfoApprovedDto`가 비슷하게 보일 수 있지만, 이 두 DTO는 서로 다른 목적과 사용 맥락을 가지고 있습니다. 이를 분리한 이유는 클린 아키텍처의 원칙을 준수하고, 확장성과 유지보수성을 높이기 위함입니다. 또한, 프론트엔드에서 요청하는 데이터 구조와 요구사항이 다르기 때문에 각 상황에 맞는 DTO를 설계하였습니다!

geonuu09 replied:
현재 제가 담당했던 CoffeeChatInfo 부분에서 공동 Dto를 사용한 부분도 있지만 추후 분리할 예정입니다.

geonuu09 committed:
fix : equals 객체비교의 문제점 발견

- 커피챗 시간 변경 시 모든 시간을 제거 후 재등록 -> PK값이 새롭게 생김.
- contains 메소드 변경 및 equals, hashCode 오버라이드 사용

geonuu09 committed:
Merge remote-tracking branch 'origin/feat/coffeechat' into feat/coffeechat

shCHO9801 reviewed (commented):
수고하셨습니다!

geonuu09 committed:
fix : private -> protected 변경
